### PR TITLE
Allow the workflow to push directly to the default branch

### PR DESCRIPTION
**Pull request by villainous-hippogriff**
For complete automation it was required that files are synced without the need for user interaction. If the option SKIP_PR is set to true (defaulted to false), it will push the changes without a PR.

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -89,7 +89,12 @@ const context = {
 	GITHUB_REPOSITORY: getVar({
 		key: 'GITHUB_REPOSITORY',
 		required: true
-	})
+	}),
+	SKIP_PR: getVar({
+		key: 'SKIP_PR',
+		required: false,
```
Comment (BetaHuhn) [inline src/config.js:95]: required defaults to false, so you can remove it here.

File: src/config.js [hunk 2]
```diff
@@ -89,7 +89,12 @@ const context = {
 	GITHUB_REPOSITORY: getVar({
 		key: 'GITHUB_REPOSITORY',
 		required: true
-	})
+	}),
+	SKIP_PR: getVar({
```
Comment (BetaHuhn) [inline src/config.js:93]: As the option only accepts true or false, specify the type like this:

```
type: 'boolean'
```

**Review by BetaHuhn — CHANGES_REQUESTED** [review]

**Comment by BetaHuhn** [comment]
Thanks for this great PR, this something I had on my mind, but haven’t implemented yet.

If you address the small changes I will merge this ASAP.

PS: Thanks for correcting my spelling mistakes too 😄

File: src/config.js (hunk 1 above)
Comment (villainous-hippogriff) [reply src/config.js]: Updated

File: src/config.js (hunk 2 above)
Comment (villainous-hippogriff) [reply src/config.js:93]: Updated

**Comment by villainous-hippogriff** [comment]
> Thanks for this great PR, this something I had on my mind, but haven’t implemented yet.
> 
> If you address the small changes I will merge this ASAP.
> 
> PS: Thanks for correcting my spelling mistakes too 😄

I updated all the changes you mentioned. The feature came really useful for automation, awesome work on the action!

**Review by BetaHuhn — APPROVED** [review]

**Comment by BetaHuhn** [comment]
> I updated all the changes you mentioned. The feature came really useful for automation, awesome work on the action!

Thanks! Will merge this later today 👍

File: src/index.js [hunk 3]
```diff
@@ -38,13 +39,16 @@ const run = async () => {
...
-			const existingPr = OVERWRITE_EXISTING_PR ? await git.findExistingPr() : undefined
-			if (existingPr && DRY_RUN === false) {
-				core.info(`Found existing PR ${ existingPr.number }`)
-				await git.setPrWarning()
+			
+			if (!SKIP_PR) {
+				await git.createPrBranch()
+
+				// Check for existing PR and add warning message that the PR maybe about to change
+				const existingPr = OVERWRITE_EXISTING_PR ? await git.findExistingPr() : undefined
```
Comment (BetaHuhn) [inline src/index.js:47]: `existingPr ` needs to be declared outside of this if statement because it is used later.

File: src/index.js [hunk 4]
```diff
@@ -118,12 +122,18 @@ const run = async () => {
 			if (hasChanges === false && modified.length < 1) {
 				core.info('File(s) already up to date')
 
-				if (existingPr) await git.removePrWarning()
+				if (SKIP_PR) {
```
Comment (BetaHuhn) [inline src/index.js:125]: This check is not really necessary, as the PR warning only gets removed if there is an existing PR.

**Comment by BetaHuhn** [comment]
I can't push anything to this PR since you made the changes on the master branch of your fork. I will merge this PR into the develop branch and fix the remaining issues there.

**Pull request merged**